lib/stacks/next.js

```diff
@@ -21,15 +21,64 @@ const install = async (projectName) => {
   return new Promise((resolve, reject) => {
     commandExists(installCmd)
       .then(() => execa(installCmd, ['install']))
-      .then(() => {
-        output.success(`Installed dependencies for ${output.cmd(projectName)}`);
+      .then(() => output.success(`Installed dependencies for ${output.cmd(projectName)}`))
+      .then(async () => {
+        if (typescript) {
+          const useYarn = installCmd === 'yarn';
+          output.info('Installing packages for TypeScript...');
+          const installDepsCmd = useYarn ? 'add' : 'install'
+          await execa(installCmd, [installDepsCmd, '-D', 'typescript', '@types/react', '@types/node']);
+          output.success('Installed dependencies for TypeScript');
+        }
         resolve();
       })
       .catch(() => reject(new Error(`${installCmd} installation failed`)));
   }).catch((error) => output.error(error.message));
 };
 
-async function next(projectName, projectPath, projectStyle) {
+const initForTypeScript = async (root) => {
+  output.info('Initializing for TypeScript...');
+
+  const jsConfigJson = await fs.readFile(sysPath.join(root, 'jsconfig.json'))
```

Comment:
We get the necessary information such as `compilerOptions.paths` from `jsconfig.json`, create the base `tsconfig.json`, and then run `next build` to merge it with the one generated by the next.js side.

The following `tsconfig.json` will be generated.

```json
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "baseUrl": ".",
    "paths": {
      "@/*": [
        "src/*"
      ]
    },
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve"
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

Reply:
I agree but I think we should let the default configuration from the nextjs team. A lot of people don't like the strict parameter as it breaks a lot of builds.